lib/datadog/statsd/origin_detection.rb

```diff
@@ -0,0 +1,156 @@
+module Datadog
+  class Statsd
```

Comment:
Should we have this under an `OriginDetection` class or something similar instead? Right now these methods are exposed and the user can call `statsd.parse_cgroup_node_path` etc